apps/crn-server/src/handlers/working-group/update-deliverables-handler.ts

```diff
@@ -0,0 +1,97 @@
+import 'source-map-support/register';
+
+import { Handler } from 'aws-lambda/handler';
+import { EventBridgeHandler } from '@asap-hub/server-common';
+import {
+  RestWorkingGroup,
+  SquidexGraphql,
+  SquidexRest,
+} from '@asap-hub/squidex';
+import { WorkingGroupDeliverable, DeliverableStatus } from '@asap-hub/model';
+import logger from '../../utils/logger';
+import { sentryWrapper } from '../../utils/sentry-wrapper';
+import { WorkingGroupEvent, WorkingGroupPayload } from '../event-bus';
+import {
+  WorkingGroupDataProvider,
+  WorkingGroupSquidexDataProvider,
+} from '../../data-providers/working-groups.data-provider';
+import { getAuthToken } from '../../utils/auth';
+import { appName, baseUrl } from '../../config';
+
+type PartialMapping<T extends string> = { [S in T]?: T };
```

Comment:
I don't _think_ so, because it requires that the keys and values are both of the defined type.

I may well have made this way too abstract given its limited use case in this file though.

I think this would work just as well.

```ts
type StatusMapping = { [S in DeliverableStatus]?: DeliverableStatus };
```

Reply:
Replaced with a more obvious type definition.